doc/nrf/protocols/matter/end_product/test_event_triggers.rst

```diff
@@ -0,0 +1,275 @@
+     For operations in the application context, use the ``Nrf::PostTask`` function.
+
+   Here is an example of how to create a callback function:
+
+   .. code-block:: c++
+
+     CHIP_ERROR MyFunctionCallback(Nrf::Matter::TestEventTrigger::TriggerValue value)
+     {
+        /* Define the required behavior of the device here. */
+
+        return CHIP_NO_ERROR;
+     }
+
+#. Register the new event trigger.
+
+   Use the following example as a guide to register a new event:
+
+   .. code-block:: c++
+
+     /* Create a new event */
+     Nrf::Matter::TestEventTrigger::EventTrigger myEventTrigger;
+
+     /* Assign all fields */
+     uint64_t myTriggerID = /* Set the trigger ID */
+     myEventTrigger.Mask = /* Fill the value mask filed */;
+     myEventTrigger.Callback = MyFunctionCallback;
+
+     /* Register the new event */
+     CHIP_ERROR err = Nrf::Matter::TestEventTrigger::Instance().RegisterTestEventTrigger(myTriggerID, myEventTrigger);
+
+     /* Remember to check the CHIP_ERROR return code */
+
+   If the returning `CHIP_ERROR` code is equal to `CHIP_ERROR_NO_MEMORY`, you need to increase the :kconfig:option:`NCS_SAMPLE_MATTER_TEST_EVENT_TRIGGERS_MAX` kconfig option to the higher value.
+
+   Here's an example to handle the ``0xF000F00000001234`` activation code, where 1234 is the event trigger value field:
+
+   .. code-block:: c++
+
+     Nrf::Matter::TestEventTrigger::EventTrigger myEventTrigger;
+     uint64_t myTriggerID = 0xF000F0000000;
+     myEventTrigger.Mask = 0xFFFF;
+     myEventTrigger.Callback = MyFunctionCallback;
+
+     CHIP_ERROR err = Nrf::Matter::TestEventTrigger::Instance().RegisterTestEventTrigger(myTriggerID, myEventTrigger);
+
+Register an existing test event trigger handler
+***********************************************
+
+In Matter SDK there are some already implemented test event trigger handlers.
+All of them inherits by `TestEventTriggerHandler` class and are implemented in various places in the Matter SDK.
+You can use them and their pre-defined behaviors using the test event triggers mechanism presented on this page.
+
+The events declared in an existing test event triggers can have the a different semantic than described in the :ref:`matter_test_event_triggers_custom_triggers` section.
+
+Use the following example as a guide to register an existing event trigger handler:
+
+.. code-block:: c++
+
+  /* Create the Trigger Handler object */
+  static TestEventTriggerHandler existingTriggerHandler;
+  CHIP_ERROR err = Nrf::Matter::TestEventTrigger::Instance().RegisterTestEventTriggerHandler(&existingTriggerHandler);
+
+  /* Remember to check the CHIP_ERROR return code */
+
+If the returning `CHIP_ERROR` code is equal to `CHIP_ERROR_NO_MEMORY`, you need to increase the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_TEST_EVENT_TRIGGERS_MAX_TRIGGERS_DELEGATES` kconfig option to the higher value.
```

Comment:
```suggestion
If the returning ``CHIP_ERROR`` code is equal to ```CHIP_ERROR_NO_MEMORY``, you need to increase the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_TEST_EVENT_TRIGGERS_MAX_TRIGGERS_DELEGATES` kconfig option to the higher value.
```